### PR TITLE
Model Optimizer incorrectly convert per-channel Quantize/Dequantize operatros to FakeQuantize

### PR DESCRIPTION
**Pull request by evolosen**
Root cause analysis: FakeQuantize nodes with constant inputs disappear after cleanup.

Solution: Set flag stop_value_propagation for such FakeQuantize

Ticket: #68029

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

**Review by vgavrilo — APPROVED** [review]

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py [hunk 1]
```diff
@@ -35,68 +35,79 @@ class QuantizeLinearResolver(MiddleReplacementPattern):
...
-                mul_high_reshape.out_port(0).connect(fake_quantize.in_port(2))
+            # if quantize_node has True attribute 'isolated', node was isolated in QuantizeDequantizeLinearResolver and
+            # will be deleted after next cleanup
+            if quantize_node.has('isolated') and quantize_node['isolated']:
+                continue
+            QuantizeLinearResolver.quantize_to_fakequantize(graph, quantize_node)
+
+    @staticmethod
+    def quantize_to_fakequantize(graph: Graph, quantize_node: Node, set_stop_value_propagation=False):
+        node_name = quantize_node.soft_get('name', quantize_node.id)
```
Comment (evolosen) [inline tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py:51]: @mvafin check this file

File: tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+from openvino.tools.mo.graph.graph import Graph, rename_nodes
+from openvino.tools.mo.middle.replacement import MiddleReplacementPattern
+from openvino.tools.mo.middle.quantize_linear_resolver import QuantizeLinearResolver
+from openvino.tools.mo.ops.const import Const
+from openvino.tools.mo.ops.reshape import Reshape
+from openvino.tools.mo.utils.error import Error
+from openvino.tools.mo.utils.graph import node_incoming_neighbourhood
+
+
+class QuantizeDequantizeLinearResolver(MiddleReplacementPattern):
```
Comment (achetver) [inline tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py:20]: Please, implement tests for this transformation

**Comment by achetver** [comment]
Please, cover new/changed functionality by tests where it is possible.

**Review by mvafin — APPROVED** [review]
LGTM, please add tests for QuantizeDequantizeLinearResolver

File: tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py (hunk 2 above)
Comment (evolosen) [reply tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py:20]: done

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py [hunk 3]
```diff
@@ -35,68 +35,79 @@ class QuantizeLinearResolver(MiddleReplacementPattern):
...
+                                               {'name': node_name + '/Mul/High'})
+        mul_low.in_port(0).get_connection().add_destination(mul_high.in_port(0))
+        mul_high.out_port(0).connect(fake_quantize.in_port(2))
+
+        cast = Cast(graph, {'dst_type': zero_point_type, 'name': node_name + '/Cast'}).create_node()
+        fake_quantize.out_port(0).connect(cast.in_port(0))
+        quantize_node.out_port(0).get_connection().set_source(cast.out_port(0))
+        rename_nodes([(quantize_node, node_name + '/TBD'), (cast, node_name)])
+
+        assert scale_y_shape is not None
```
Comment (achetver) [inline tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py:99]: Please, add message to the assert

File: tools/mo/unit_tests/mo/middle/quantize_dequantize_linear_resolver_test.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    **regular_op_with_empty_data('non_const_mul_high', {'op': 'Mul'}),
+    **regular_op_with_empty_data('non_const_fq', {'op': 'FakeQuantize'}),
+    **regular_op_with_empty_data('non_const_cast', {'op': 'Cast', 'type': 'Convert', 'dst_type': np.uint8}),
+    **result('result'),
+
+}
+
+
+class QuantizeDequantizeLinearResolverTest(unittest.TestCase):
+    def test_group_norm_1(self):
```
Comment (achetver) [inline tools/mo/unit_tests/mo/middle/quantize_dequantize_linear_resolver_test.py:53]: I think the name of the test isn't correct :)

Please change it.

**Review by achetver — APPROVED** [review]
In general looks good. Please address comments above.

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py [hunk 5]
```diff
@@ -35,68 +35,80 @@ class QuantizeLinearResolver(MiddleReplacementPattern):
...
-                                                               {'name': node_name + '/Reshape/Mul/high'})
-
-                fake_quantize.in_port(1).get_connection().set_destination(mul_low_reshape.in_port(0))
-                fake_quantize.in_port(2).get_connection().set_destination(mul_high_reshape.in_port(0))
-
-                mul_low_reshape.out_port(0).connect(fake_quantize.in_port(1))
-                mul_high_reshape.out_port(0).connect(fake_quantize.in_port(2))
+            # if quantize_node has True attribute 'isolated', node was isolated in QuantizeDequantizeLinearResolver and
+            # will be deleted after next cleanup
+            if quantize_node.has('isolated') and quantize_node['isolated']:
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py:45]:
```suggestion
            if quantize_node.has_and_set('isolated'):
```

File: tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                                            dq_zerop.soft_get('type') == 'Const'
+                scales_and_zerop_equals = np.array_equal(q_scale.value, dq_scale.value) and \
+                                          np.array_equal(q_zerop.value, dq_zerop.value)
+                # only constant as for zero_point/scale supported
+                # only patterns with same scale/zero_point values for Q and DQ are supported
+                if not (scales_and_zerop_is_const or scales_and_zerop_equals):
+                    continue
+
+                QuantizeLinearResolver.quantize_to_fakequantize(graph, quantize_node, True)
+                quantize_node['isolated'] = True
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py:66]: can we simply delete this fq?

File: tools/mo/openvino/tools/mo/middle/dequantize_linear_resolver.py [hunk 7]
```diff
@@ -33,11 +37,13 @@ def find_and_replace_pattern(self, graph: Graph):
             model_data_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
             cast = Cast(graph, {'dst_type': model_data_type, 'name': node_name + '/Cast'}).create_node()
             dequantize_node.in_port(0).get_connection().set_destination(cast.in_port(0))
-            mul = Mul(graph, {}).create_node()
+            mul = Mul(graph, {'can_be_fused':False}).create_node()
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/dequantize_linear_resolver.py:40]: pep8

File: tools/mo/openvino/tools/mo/front/sub.py [hunk 8]
```diff
@@ -42,4 +42,6 @@ def sub_to_add_replacement(sub: Node):
 
     def find_and_replace_pattern(self, graph: Graph):
         for sub in graph.get_op_nodes(op='Sub'):
+            if sub.has('dont_replace') and sub['dont_replace']:
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/front/sub.py:45]: please leave a comment here about quantization and rename attribute name to smth like `zero_point_sub`

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py [hunk 9]
```diff
@@ -35,68 +35,80 @@ class QuantizeLinearResolver(MiddleReplacementPattern):
...
+            output_high_value = 255.0
+        else:
+            raise Error('Not expected type {} for zero point value in node {}'.format(
+                zero_point_type, zerop.soft_get('name')))
+
+        fake_quantize = create_op_with_const_inputs(graph, FakeQuantize, {3: float_array(output_low_value),
+                                                                          4: float_array(output_high_value)},
+                                                    {'levels': 256, 'name': node_name + '/FakeQuantize'})            
+        if set_stop_value_propagation:
+            fake_quantize['not_compress'] = True
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py:78]: nit: `stop_compression`?

File: tools/mo/openvino/tools/mo/front/sub.py (hunk 8 above)
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/front/sub.py:45]:
```suggestion
            if sub.has_and_set('dont_replace'):
```

**Review by jane-intel — CHANGES_REQUESTED** [review]

File: tools/mo/openvino/tools/mo/front/sub.py (hunk 8 above)
Comment (evolosen) [reply tools/mo/openvino/tools/mo/front/sub.py]: done

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py (hunk 5 above)
Comment (evolosen) [reply tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py]: done

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py [hunk 10]
```diff
@@ -35,68 +35,80 @@ class QuantizeLinearResolver(MiddleReplacementPattern):
...
-                                                               {'name': node_name + '/Reshape/Mul/high'})
-
-                fake_quantize.in_port(1).get_connection().set_destination(mul_low_reshape.in_port(0))
-                fake_quantize.in_port(2).get_connection().set_destination(mul_high_reshape.in_port(0))
-
-                mul_low_reshape.out_port(0).connect(fake_quantize.in_port(1))
-                mul_high_reshape.out_port(0).connect(fake_quantize.in_port(2))
+            # if quantize_node has True attribute 'isolated', node was isolated in QuantizeDequantizeLinearResolver and
+            # will be deleted after next cleanup
+            if quantize_node.has_and_set('isolated'):
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py:45]:
```suggestion
            if quantize_node.has_and_set('isolated'):  # node is detached and will be eliminated during the next clean up
```

File: tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['layout'] == 'NCHW']
+
+    def run_after(self):
+        from openvino.tools.mo.middle.quantize_fuses import MarkNodesToFuseUpToFakeQuantize
+        return [MarkNodesToFuseUpToFakeQuantize]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for dequantize_node in graph.get_op_nodes(op='DequantizeLinear'):
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py:34]: why this is not a matcher pass?

**Review by jane-intel — COMMENTED** [review]
Looks good. Could you please answer my questions?

File: tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py (hunk 11 above)
Comment (evolosen) [reply tools/mo/openvino/tools/mo/middle/quantize_dequantize_linear_resolver.py:34]: I thought it is oldstyle to use matcher

File: tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py (hunk 10 above)
Comment (evolosen) [reply tools/mo/openvino/tools/mo/middle/quantize_linear_resolver.py]: done

**Review by jane-intel — APPROVED** [review]

**Pull request merged**